### ISSUE TITLE
Feature Request: Always discard project state when "Restore Previous Windows On Start" is false.

### ISSUE THREAD

**Issue by phyllisstein** [atom/atom#14303]
### Prerequisites
* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Hey Atomizers! I wanted to kick a little feature request your way, if it's not too much trouble. I've noticed that in the beta build of Atom, reopening a project in a window with a preexisting tab gives one the option to discard Atom's autosaved project state. I was wondering if it wouldn't be more consistent with the "Restore Previous Windows on Start" setting if, when "restore windows" was flagged `false`, opening any project---from the menu, via drag-and-drop, from the CLI, from a project management package---automatically dropped its state on the floor.

I'm ultimately trying to get back to a world like the one I enjoyed when setting `hot_exit` to `false` in Sublime Text, wherein projects never remember expanded directories or opened tabs. (Unsaved buffers are another story: losing those was a part of the deal with Sublime, but Atom's behavior could in theory be gentler there.) Nineish times out of ten, rather than dropping me helpfully where I was working six minutes ago, the state restoration drops me annoyingly where I was working six weeks ago; it's easier just to get back to where I was on my own.

The behavior is _so close_ to what I'm looking for in `1.17.0-beta3`, _except_ that reopening a project in a clean window always restores its tabs (which focus themselves in the Tree View as I close them, ack):

![screenflow](https://cloud.githubusercontent.com/assets/2207980/25434235/84b25d5e-2a59-11e7-9d2e-4a45d685fadc.gif)

Can't imagine this being super-high-priority by any stretch, but I also don't quite have the Atom-fu to tackle it on my own. Would be grateful if someone looking for an easy PR would consider it. :pray:

### Steps to Reproduce
1. Open a whole mess'a tabs in an Atom project window.
2. Close the project without explicitly closing your tabs.
3. Open a new window and reopen that selfsame project.
4. Tabs. Why did it have to be tabs.

**Suggested behavior:** When "Restore Previous Windows on Start" is disabled, Atom should discard all preexisting project state---including opened tabs.

**Actual behavior:** Atom selectively remembers some preexisting project state regardless of the setting---including opened tabs.

**Reproduces how often:** 100%

### Versions
```
❯ atom --version
Atom    : 1.17.0-beta3
Electron: 1.3.14
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

```
❯ apm --version
apm  1.17.0
npm  3.10.5
node 6.9.5 x64
python 2.7.13
git 2.12.2
```

```
❯ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 10.12.4 (16E195)
      Kernel Version: Darwin 16.5.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: Zbornak
      User Name: Daniel P. Shannon (daniel)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Disabled
      Time since boot: 1 day 12:19
```

### Additional Information
💖

**Comment by rsese** [comment]
> 1. Open a whole mess'a tabs in an Atom project window.
> 2. Close the project without explicitly closing your tabs.

Hey @phyllisstein :wave:  Just to clarify step 2 (I couldn't quite follow it in the GIF),  how did you close the project with closing the tabs opened in step 1?

**Comment by phyllisstein** [comment]
Sure! I mashed <key>⌘</key>+<key>Shift</key>+<key>W</key>. It looks like the behavior is similar when using the native red "Close" bubble: project reopens with a collapsed Tree View and all its old tabs.

**Comment by narcolepticinsomniac** [comment]
>I'm ultimately trying to get back to a world like the one I enjoyed when setting hot_exit to false in Sublime Text

I've been googling this exactly, trying to figure out how to accomplish this in Atom. Many editors still use this as the default behavior, but I've never seen one that didn't least least provide the option in some form. Sublime and VSC both have similar hot exit JSON settings to tweak.

If I have Atom set to open to an empty editor, why would I want it to save edits internally? Opening files later prompts a load/discard confirmation for edits which I don't even recall because I never intended them to be saved internally. It saves it as  a "project", so other tabs that were randomly open are reopened.

I get that you guys have put a lot of effort into the hot exit, so edits save internally as projects. It seems to work pretty well, and I understand it being the default as it is in many modern editors, but every editor should be capable of switching to a traditional save/discard prompt for all open, edited files.

**Comment by Zangoran** [comment]
@rsese This is still an issue. Is this on any road map? Thanks for you great work!

**Comment by narcolepticinsomniac** [comment]
Loved Atom for it's customizability, but quit using it because pf the inability to set a classic save prompt and do away with random unintentional unsaved changes, not to mention random "projects" opening when opening a simple file, because those files happened to be open at the same time and are now forever associated. Would love to go back to Atom, but not until there's a simple hot-exit option.

**Comment by Zangoran** [comment]
@narcolepticinsomniac Yeah its very annoying. But I still use Atom anyway, lol.

**Comment by richardbushell** [comment]
Yes, I agree with the above, please provide a simple tick box in the config to close tabs on exit. Obviously it would need to provide the standard prompt to save any files with changes, and close the others automatically). This is a deal breaker for me, it's very annoying.
